libs/ml/include/ml/ops/ops.hpp

```diff
@@ -51,6 +53,17 @@ class Ops
    * in ASSERT. On Forward you can use output.shape() and on Backward there is error_signal.shape()
    */
   virtual std::vector<SizeType> ComputeOutputShape(VecTensorType const &inputs) const = 0;
+  std::vector<SizeType>         ComputeDefaultOutputShape(VecShapesType const &input_shapes)
```

Comment:
i'm not sure about your naming - is 'default' really the right word for all these functions?

In any case please add function comment with description

Reply:
Not sure "default" is the optimal one, the intention was to show it is not the "working" shape, e.g. "default" ignores batch-size of input data and always have 1 as a trailing dimension. It could be also smth like "single", "minimal", "primitive", so I am open to advice here :)

Reply:
I've changed it to SliceOutputShape - where a "slice" is 1 data point ("slice") from an input batch.